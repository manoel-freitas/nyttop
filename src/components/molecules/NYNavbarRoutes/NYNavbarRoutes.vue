<template>
  <ul :class="listClass">
    <ny-navbar-item
      v-for="(route, index) in routes"
      :key="index"
      :item-class="listItemClass"
      :to="route.to"
      :label="route.label"
    ></ny-navbar-item>
  </ul>
</template>

<script>
export default {
  props: {
    listClass: {
      type: String,
      required: true
    },
    listItemClass: {
      type: String,
      required: true
    }
  },
  data() {
    return {
      routes: [
        {
          to: {
            path: '/'
          },
          label: 'Home'
        },
        {
          to: {
            path: '/section/sci-tech'
          },
          label: 'Science and Tech'
        },
        {
          to: {
            path: '/section/health'
          },
          label: 'Health'
        },
        {
          to: {
            path: '/section/politics'
          },
          label: 'Politics'
        },
        {
          to: {
            path: '/section/world'
          },
          label: 'World'
        }
      ]
    }
  }
}
</script>
