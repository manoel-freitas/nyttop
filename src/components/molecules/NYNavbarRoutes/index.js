import NYNavbarRoutes from './NYNavbarRoutes'

export default NYNavbarRoutes
