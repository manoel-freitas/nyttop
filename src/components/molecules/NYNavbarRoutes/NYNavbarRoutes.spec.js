/// <reference types="jest" />

import { shallowMount } from '@vue/test-utils'
import NYNavbarRoutes from './NYNavbarRoutes.vue'

const nyNavbarRoutes = shallowMount(NYNavbarRoutes, {
  props: {
    listClass: 'levelbar',
    listItemClass: 'levelbar-item'
  },
  stubs: ['ny-navbar-item']
})
test('NYNavbarRoutes has to be mounted', () => {
  expect(nyNavbarRoutes.isVueInstance()).toBeTruthy()
})

test('snapshot test', () => {
  expect(nyNavbarRoutes.element).toMatchSnapshot()
})
