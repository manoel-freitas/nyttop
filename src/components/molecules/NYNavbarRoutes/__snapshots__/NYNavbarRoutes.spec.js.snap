// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<ul>
  <ny-navbar-item-stub
    label="Home"
    to="[object Object]"
  />
  <ny-navbar-item-stub
    label="Science and Tech"
    to="[object Object]"
  />
  <ny-navbar-item-stub
    label="Health"
    to="[object Object]"
  />
  <ny-navbar-item-stub
    label="Politics"
    to="[object Object]"
  />
  <ny-navbar-item-stub
    label="World"
    to="[object Object]"
  />
</ul>
`;
